### PR TITLE
feat: allow specific description for CLI options

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
Feature

 - add `cliDescription` -  Sometimes we need specific descriptions for CLI options, for example when we are deprecating a CLI option only in favor of another  
   - https://github.com/webpack/webpack-dev-server/pull/4091 
   - https://github.com/webpack/webpack-dev-server/pull/4098
 
 - add `negatedDescription` - `webpack-cli` creates a negative flag for each `boolean` flag, it uses [`negatedDescription` property for description if available](https://github.com/webpack/webpack-cli/blob/ebf665344079489f2b4f36d7d1c5e46b36550985/packages/webpack-cli/lib/webpack-cli.js#L467).

**Did you add tests for your changes?**
Yes

**Does this PR introduce a breaking change?**
No

**What needs to be documented once your changes are merged?**

It is for internal usage, nothing to be documented.

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -97,10 +98,23 @@ const getArguments = (schema = webpackSchema) => {
 	const getDescription = path => {
 		for (const { schema } of path) {
 			if (schema.cli && schema.cli.helper) continue;
+			if (schema.cliDescription) return schema.cliDescription;
```
Comment (sokra) [inline lib/cli.js:101]: since this is cli specific this should be moved into `schema.cli.description`

File: lib/cli.js [hunk 2]
```diff
@@ -97,10 +98,23 @@ const getArguments = (schema = webpackSchema) => {
...
 
+	/**
+	 *
+	 * @param {PathItem[]} path path in the schema
+	 * @returns {string | undefined} negative description
+	 */
+	const getNegatedDescription = path => {
+		for (const { schema } of path) {
+			if (schema.cli && schema.cli.helper) continue;
+			if (schema.negatedDescription) return schema.negatedDescription;
```
Comment (sokra) [inline lib/cli.js:114]:
```suggestion
			if (schema.cli.negatedDescription) return schema.cli.negatedDescription;
```

**Review by sokra — CHANGES_REQUESTED** [review]

File: lib/cli.js (hunk 1 above)
Comment (snitin315) [reply lib/cli.js]: Makes sense 👍🏻

**Comment by snitin315** [comment]
@sokra, thanks for the update.

**Comment by sokra** [comment]
Thanks

**Pull request merged**